download_weather_streaming.py

```
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+"""
+Streaming weather data download - saves data progressively to avoid memory issues
+"""
+
+import asyncio
+import csv
```

Ruff (0.12.2)

`csv` imported but unused

Remove unused import: `csv`

(F401)